server/utils/dateUtils.ts

```diff
@@ -1,32 +1,50 @@
 import { findTimeZone, getZonedTime } from 'timezone-support'
 
-function getMonthName(num: number): string {
-  const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec']
-  return months[num - 1]
-}
-
-function formatDateTimeOrEmptyString(d: string | null): string {
-  if (d === null) {
-    return ''
+export default class DateUtils {
+  static getMonthName(num: number): string {
+    const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec']
+    return months[num - 1]
   }
 
-  const date = new Date(d)
+  static formatDateTimeOrEmptyString(d: string | null): string {
+    if (d === null) {
+      return ''
+    }
+
+    const date = new Date(d)
+
+    if (Number.isNaN(Number(date))) {
+      return ''
+    }
+
+    const ukDate = getZonedTime(date, findTimeZone('Europe/London'))
+
+    const { year } = ukDate
+    const month = DateUtils.getMonthName(ukDate.month)
+    const day = `0${ukDate.day}`.slice(-2)
+    const hour = `0${ukDate.hours}`.slice(-2)
+    const minutes = `0${ukDate.minutes}`.slice(-2)
 
-  if (Number.isNaN(Number(date))) {
-    return ''
+    return `${day} ${month} ${year}, ${hour}:${minutes}`
   }
 
-  const ukDate = getZonedTime(date, findTimeZone('Europe/London'))
+  static getDateFromFormattedDateTime(dateTime: string | null): string {
+    const formattedDateTime = DateUtils.formatDateTimeOrEmptyString(dateTime)
 
-  const { year } = ukDate
-  const month = getMonthName(ukDate.month)
-  const day = `0${ukDate.day}`.slice(-2)
-  const hour = `0${ukDate.hours}`.slice(-2)
-  const minutes = `0${ukDate.minutes}`.slice(-2)
+    if (formattedDateTime === '') {
+      return formattedDateTime
+    }
 
-  return `${day} ${month} ${year}, ${hour}:${minutes}`
-}
+    return formattedDateTime.split(',')[0].trim()
```

Comment:
Personally I'd probably have made these two separate methods that return the components, and then have a method which combines them, instead of returning the combined thing and then splitting it back out again, but I don't think it matters enough to change.

Reply:
This is a very good point, I'm changing this as it'll also make it easier to update in future and change to 12hr format etc.